lsst_efd_client/auth_helper.py

```diff
@@ -4,65 +4,42 @@
 import os
 import stat
 import yaml
+import requests
+from urllib.parse import urljoin
 
 
 class NotebookAuth:
     """Class to help keep authentication credentials secret.
 
-    Credentials can be passed directly in a dictionary, the name of an
-    environment variable that contains the path to the credentials file may be
-    passed, or a path to the file may be passed directly.  The credential
-    location is checked in that order.
+    Credentials can be retrieved either from a service endopint or 
+    from a file on disk.  The credential location is checked in that order.
 
     Parameters
     ----------
+    service_endpoint : `str`, optional
+        Endopint of the service to use for credentials.
+        (https://roundtable.lsst.codes/segwarides/ by default)
     path : `str`, optional
-        Path to thuse when reading credentials from disk
+        Path to use when reading credentials from disk
         ('~/.lsst/notebook_auth.yaml' by default).
-    env_var : `str`, optional
-        Name of an environment variable to check for the
-        path to the credentials file (`None` by default).
-    auth_dict : `dict`, optional
-        Dictionary of dictionaries of credentials.  Each key is the name of
-        a potential authenticator.  The credential dictionaries must have the
-        following keys: 'host', 'username', and 'password'.  This is None by
-        default.
 
     Raises
     ------
     ValueError
-       Raised if ``auth_dict``, ``env_var``, and ``path`` are all `None`.
+       Raised if the auth file is absent or if ``path`` and ``service_endpoint``
+       are both `None`.
     IOError
        Raised if either the credentials file has the wrong permissions or
        if the file fails to load.
     """
 
-    def __init__(self, path='~/.lsst/notebook_auth.yaml', env_var=None, auth_dict=None):
-        if auth_dict is not None:
-            self.auth_dict = auth_dict
-            return
-        if env_var is not None and env_var in os.environ:
-            secret_path = os.path.expanduser(os.environ[env_var])
-        elif path is None:
-            raise ValueError(
-                "No default path provided to auth file")
+    def __init__(self, service_endpoint="https://roundtable.lsst.codes/segwarides/", path='~/.lsst/notebook_auth.yaml'):
```

Comment:
Make sure you remove the `path` parameter here and from the docstring.

Reply:
OK.  Thanks for catching that.  Will do.